run/image-processing/src/main/java/com/example/cloudrun/Body.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.cloudrun;
+
+// Body.Message is the payload of a Pub/Sub event. Please refer to the docs for
+// additional information regarding Pub/Sub events.
+public class Body {
+
+  private Message message;
+
+  public Body() {}
+
+  public Message getMessage() {
+    return message;
+  }
+
+  public void setMessage(Message message) {
+    this.message = message;
+  }
+
+  public class Message {
+
+    private String messageId;
+    private String publishTime;
+    private String data;
+
+    public Message() {}
+
+    public Message(String messageId, String publishTime, String data) {
+      this.messageId = messageId;
+      this.publishTime = publishTime;
+      this.data = data;
+    }
+
+    public String getMessageId() {
+      return messageId;
+    }
+
+    public void setMessageId(String messageId) {
+      this.messageId = messageId;
+    }
+
+    public String getPublishTime() {
+      return publishTime;
+    }
+
+    public void setPublishTime(String publishTime) {
+      this.publishTime = publishTime;
+    }
+
+    public String getData() {
+      return data;
+    }
+
+    public void setData(String data) {
+      this.data = data;
+    }
+  }
```

Comment:
See [Effective Java 3rd Ed.](https://www.oreilly.com/library/view/effective-java-3rd/9780134686097/) Chapter 3 -- ask me to send you a link.

Reply:
Since it's springboot, there is probably some annotation that can inject these methods for you.

Reply:
This is the cleanest way from what I can tell. Do I need to implement equals, toString, etc?

Reply:
No, though I will usually point it out until you do.  Since you are using SpringBoot, you just need to put one or two lines of annotation at the top of the class and it will create them for you.